src/Sulu/Bundle/PageBundle/Resources/js/containers/Form/fields/SettingsVersions.js

```diff
@@ -40,6 +40,75 @@ class PageSettingsVersions extends React.Component<Props> {
         });
     }
 
+    @computed get resourceKey(): string {
+        const {
+            schemaOptions: {
+                resource_key: {
+                    value: resourceKey,
+                } = {},
+            },
+        } = this.props;
+
+        if (resourceKey === undefined || typeof resourceKey !== 'string') {
+            throw new Error(
+                'The "resource_key" schemaOption is mandatory and must be a string, but received ' +
+                typeof resourceKey + '!'
+            );
+        }
+
+        return resourceKey;
+    }
+
+    @computed get listKey(): string {
+        const {
+            schemaOptions: {
+                list_key: {
+                    value: listKey,
+                } = {},
+            },
+        } = this.props;
+
+        if (listKey === undefined || typeof listKey !== 'string') {
+            throw new Error(
+                'The "list_key" schemaOption is mandatory and must be a string, but received ' +
+                typeof listKey + '!'
+            );
+        }
+
+        return listKey;
+    }
+
+    @computed get userSettingsKey(): string {
+        const {
+            schemaOptions: {
+                user_settings_key: {
+                    value: userSettingsKey,
+                } = {},
+            },
+        } = this.props;
+
+        if (userSettingsKey === undefined || typeof userSettingsKey !== 'string') {
+            throw new Error(
```

Comment:
Same idea as above, what about a fallback (probably to the `listKey` in this case).